crates/wasi-nn/src/backend/winml.rs

```diff
@@ -175,23 +158,19 @@ impl BackendExecutionContext for WinMLExecutionContext {
         if let Some(result) = &self.result {
             let output_features = self.session.Model()?.OutputFeatures()?;
             let index = self.find(id, &output_features)?;
-            let output = output_features.GetAt(index)?;
-            // TODO: this only handles FP32!
-            let tensor = result
-                .Outputs()?
-                .Lookup(&output.Name()?)?
-                .cast::<TensorFloat>()?;
-            let dimensions = dimensions_as_u32(&tensor.Shape()?)?;
-            let view = tensor.GetAsVectorView()?;
-            let mut data = Vec::with_capacity(view.Size()? as usize * size_of::<f32>());
-            for f in view.into_iter() {
-                data.extend(f.to_le_bytes());
-            }
-            Ok(Tensor {
-                ty: TensorType::Fp32,
-                dimensions,
-                data,
-            })
+            let output_feature = output_features.GetAt(index)?;
+            let tensor_kind = match output_feature.Kind()? {
+                windows::AI::MachineLearning::LearningModelFeatureKind::Tensor => output_feature
+                    .cast::<TensorFeatureDescriptor>()?
+                    .TensorKind()?,
+                _ => unimplemented!(),
+            };
+            // TODO: this only handles FP16, FP32 and I64!
```

Comment:
This kind of comment can be moved from here to the `unimplemented!()` at the end of `to_tensor` so that both users and code readers can take advantage of this information.